### PR TITLE
chore: Bump version to 0.3.1

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:*

*Description of changes:*

- bump `amplify-ios` version to [1.18.3](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.18.3)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_api/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ The Amplify Flutter API category plugin.
 
 ### Visit our [Web Site](https://docs.amplify.aws/) to learn more about AWS Amplify.
 
-## Changes for version 0.3.0-rc.1
+## Changes for version 0.3.0 and above
```
Comment (dnys1) [inline packages/amplify_api/README.md:9]: Good catch, thanks!

**Review by dnys1 — APPROVED** [review]
Thanks!

**Review by ragingsquirrel3 — APPROVED** [review]

File: packages/amplify_storage_s3/android/build.gradle [hunk 2]
```diff
@@ -28,7 +28,7 @@ apply from: './coverage.gradle'
 def amplifyCore = findProject(':amplify_core')
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 31
```
Comment (haverchuck) [inline packages/amplify_storage_s3/android/build.gradle:31]: Is this necessary?

**Review by haverchuck — APPROVED** [review]
LGTM, other than comment on storage/build.gradle.

**Pull request merged**